tests/js_test.go

```diff
@@ -303,6 +304,99 @@ func TestInternalizeDate(t *testing.T) {
 	}
 }
 
+func TestInternalizeStruct(t *testing.T) {
+	type Person struct {
+		Name string
+		Age  int
+	}
+	var a, expected Person
+	expected = Person{Name: "foo", Age: 952}
+
+	js.Global.Set("return_person", func(p *Person) *Person {
+		if p == nil {
+			t.Fail()
+			return nil
+		}
+		a = *p
+		return p
+	})
+
+	js.Global.Call("eval", "return_person({Name: 'foo', Age: 952})")
+	if diff := cmp.Diff(a, expected); diff != "" {
+		t.Errorf("Mismatch (-want +got):\n%s", diff)
+	}
+}
+func TestInternalizeStructUnexportedFields(t *testing.T) {
+	type Person struct {
+		Name string
+		age  int
+	}
+	var a, expected Person
+	expected = Person{Name: "foo", age: 0}
+	js.Global.Set("return_person", func(p *Person) *Person {
+		a = *p
+		return p
+	})
+
+	js.Global.Call("eval", "return_person({Name: 'foo', age: 952})")
+
+	// Manually check unexported fields
+	if a.age != expected.age {
+		t.Errorf("Mismatch in age: got %v, want %v", a.age, expected.age)
+	}
```

Comment:
You can use https://pkg.go.dev/github.com/google/go-cmp/cmp#AllowUnexported make diff compare both exported and unexported fields.

Reply:
yea, so I've already tried it, but I got 
`Error: no support for forcibly accessing unexported fields` 

full func
```go

func TestInternalizeStructUnexportedFields(t *testing.T) {
	type Person struct {
		Name string
		age  int
	}
	var a, expected Person
	expected = Person{Name: "foo", age: 0}

	js.Global.Set("return_person", func(p *Person) *Person {
		a = *p
		return p
	})

	js.Global.Call("eval", "return_person({Name: 'foo', age: 952})")

	// Using cmp.Diff with AllowUnexported to compare both exported and unexported fields
	if diff := cmp.Diff(a, expected, cmp.AllowUnexported(Person{})); diff != "" {
		t.Errorf("Mismatch (-want +got):\n%s", diff)
	}
}
```

Reply:
That's odd, I'm pretty sure I was able to use it with private fields in another project. In any case, this is too much of a minor issue to block this PR on.